openlibrary/plugins/worksearch/languages.py

```diff
@@ -70,7 +70,12 @@ class index_json(delegate.page):
 
     @jsonapi
     def GET(self):
-        return json.dumps(get_top_languages(15))
+        i = web.input(limit=None)
+        if i.limit is not None:
+            limit = safeint(i.limit)
+            return json.dumps(get_top_languages(limit or 15))
+        else:
+            return json.dumps(get_top_languages(15))
```

Comment:
```suggestion
        i = web.input(limit=15)
        return json.dumps(get_top_languages(safeint(i.limit, 15)))
```
What about something like this? There are always trade offs everywhere, but I am hopeful this isn't so many nested functions as to be confusing. Thoughts?

Also, good call on `safeint`.

Reply:
Yeah I think that's a good bit cleaner. Thanks for the suggestion! I'll go ahead and commit that.